### PR TITLE
[Micas/Platform]platform support M2-W6510-32C

### PR DESCRIPTION
**Pull request by philo-micas**
#### Why I did it
Add new platform m2-w6510-32c(Trident 3)
ASIC Vendor: Broadcom
Switch ASIC: Trident 3
Port Config: 32×100G QSFP28

##### Work item tracking
- Microsoft ADO **(number only)**:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by philo-micas** [comment]
[report_m2_32c.zip](https://github.com/sonic-net/sonic-buildimage/files/12377887/report_m2_32c.zip)
@yxieca hi xieying, help assign one person to review code, thanks very much!

**Comment by yxieca** [comment]
@philo-micas please if the semgrep check failure. Prince can review after the checker passed.

**Comment by philo-micas** [comment]
> @philo-micas please if the semgrep check failure. Prince can review after the checker passed.

@yxieca @prgeor  Just ignore the semgrep error about `xml` since 'defusedxml' has not isntalled in sonic currentlly, help review, thanks!

**Comment by philo-micas** [comment]
@prgeor since this pr's compilation depends on https://github.com/sonic-net/sonic-buildimage/pull/15888/, so when you review, just skip 48v8c's code, only focus on 32c

File: device/micas/x86_64-micas_m2-w6510-32c-r0/monitor.py [hunk 1]
```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/python3
+#   * onboard temperature sensors
+#   * FAN trays
+#   * PSU
+#
+import os
+import xml.etree.ElementTree as ET
```
Comment (maipbui) [inline device/micas/x86_64-micas_m2-w6510-32c-r0/monitor.py:7]: >xml [](http://example.com/codeflow?start=7&length=3)

suggest use lxml library to fix Semgrep alert, you may check similar fix [https://github.com/sonic-net/sonic-buildimage/pull/11779/files](https://github.com/sonic-net/sonic-buildimage/pull/11779/files)

Comment (philo-micas) [reply device/micas/x86_64-micas_m2-w6510-32c-r0/monitor.py]: > > xml
> 
> suggest use lxml library to fix Semgrep alert, you may check similar fix https://github.com/sonic-net/sonic-buildimage/pull/11779/files

thanks! done!

**Comment by philo-micas** [comment]
> @philo-micas please if the semgrep check failure. Prince can review after the checker passed.

done!

**Comment by philo-micas** [comment]
[Uploading Platform Test Report-M2-W6510-48V8C-0103.docx…]()
Kernel 6.1.38 adaption Platform test report from sonic-mgmt.

**Comment by philo-micas** [comment]
@lguohan  Hi guohan, we've completed the adaptation for kernel 6.1.38, please help assign a code review to forward this PR, thanks!

**Comment by philo-micas** [comment]
@lguohan Have done kernel 6.1 adaptation, as we have reviewed and fixed all reviewed comments month ago, (leave one semgrep error as 'defusedxml' not installed in sonic), help push this pr forward again, thanks.

**Comment by philo-micas** [comment]
[report_M2-W6510-32C_0112.zip](https://github.com/sonic-net/sonic-buildimage/files/13925175/report_M2-W6510-32C_0112.zip)
Kernel 6.1.38 adaption
Platform test report from sonic-mgmt.

**Comment by philo-micas** [comment]
> [report_M2-W6510-32C_0112.zip](https://github.com/sonic-net/sonic-buildimage/files/13925175/report_M2-W6510-32C_0112.zip) Kernel 6.1.38 adaption Platform test report from sonic-mgmt.

@lguohan Hi guohan, test report has been submitted, please help to push forward this pr. Thanks.

**Comment by StephenSJCA** [comment]
> > [report_M2-W6510-32C_0112.zip](https://github.com/sonic-net/sonic-buildimage/files/13925175/report_M2-W6510-32C_0112.zip) Kernel 6.1.38 adaption Platform test report from sonic-mgmt.
> 
> @lguohan Hi guohan, test report has been submitted, please help to push forward this pr. Thanks.

@lguohan @prgeor @yxieca **Hi Guohan, Prince, and Ying, please take the time to push forward this PR, it is urgent for us, many thanks!**

**Comment by philo-micas** [comment]
Leave one semgrep error as 'defusedxml' was not installed in sonic.

**Comment by philo-micas** [comment]
Please use new PR link: https://github.com/sonic-net/sonic-buildimage/pull/20625

**Pull request closed without merging**